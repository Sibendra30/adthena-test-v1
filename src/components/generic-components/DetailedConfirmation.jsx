import React, { useState } from 'react';

function DetailedConfirmation({
  headingText,
  contentText,
  deleteBtnTxt = 'Delete All',
  deleteAllHandler,
  cancelBtnText = 'Cancel',
  cancelHandler,
  detailedView,
  confirmInstr = 'Confirm Instr'
}) {
  const [enableDelete, setEnableDelete] = useState(false);
  const [textValue, setTextValue] = useState('');
  const onTextChange = (e) => {
    setTextValue(e.target.value);
    if (e.target.value === 'Delete') {
      setEnableDelete(true);
    }
    if (enableDelete && e.target.value !== 'Delete') {
      setEnableDelete(false);
    }
  };
  console.log(enableDelete);
  return (
    <div>
      <h5>{headingText}</h5>
      <div className="dtl-confirmation-body">{contentText}</div>
      <div className="confirmation-details">{detailedView}</div>
      <div className="confirmation-instr">{confirmInstr}</div>
      <input type="text" value={textValue} onChange={onTextChange} />
      <div className="btn-wrapper">
        <button className="btn btn-primary" disabled={enableDelete ? '' : 'disabled'} onClick={deleteAllHandler}>{deleteBtnTxt}</button>
        <button className="btn btn-secondary" onClick={cancelHandler}>{cancelBtnText}</button>
      </div>
    </div>
  );
}

export default DetailedConfirmation;
