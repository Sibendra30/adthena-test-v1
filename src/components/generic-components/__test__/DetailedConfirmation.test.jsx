import React from 'react';
import { render } from '@testing-library/react';
import DetailedConfirmation from '@components/generic-components/DetailedConfirmation';

describe('Testing Detailed Confirmation Component', () => {
  let containerComp;
  beforeEach(() => {
    const { container } = render(<DetailedConfirmation />);
    containerComp = container;
  });
  it('Detailed Confirmation component should render', () => {
    expect(containerComp).toMatchSnapshot();
  });
  it('Should load two buttons', () => {
    expect(containerComp.querySelectorAll('button').length).toEqual(2);
  });
  it('Should load delete text box', () => {
    expect(containerComp.querySelectorAll('input[type="text"]').length).toEqual(1);
  });
  it('Should load confirmation header, body & history', () => {
    expect(containerComp.querySelectorAll('.confirm-warning-header').length).toEqual(1);
    expect(containerComp.querySelectorAll('.confirmation-details').length).toEqual(1);
    expect(containerComp.querySelectorAll('.confirmation-instr').length).toEqual(1);
  });
});
