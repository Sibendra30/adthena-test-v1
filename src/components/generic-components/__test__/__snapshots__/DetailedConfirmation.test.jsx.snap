// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Detailed Confirmation Component Detailed Confirmation component should render 1`] = `
<div>
  <div>
    <div
      class="confirm-warning-header"
    >
      <img
        alt="warning"
        src="/images/alarm.png"
      />
      <h5 />
    </div>
    <div
      class="dtl-confirmation-body"
    />
    <div
      class="confirmation-details"
    />
    <div
      class="confirmation-instr"
    >
      Confirm Instr
    </div>
    <input
      type="text"
      value=""
    />
    <div
      class="btn-wrapper"
    >
      <button
        class="btn btn-primary"
        disabled=""
      >
        Delete All
      </button>
      <button
        class="btn btn-secondary"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
